cpp/src/arrow/c/bridge_test.cc

```diff
@@ -4302,6 +4302,16 @@ TEST_F(TestDeviceArrayRoundtrip, Primitive) {
   TestWithJSON(mm, int32(), "[4, 5, null]");
 }
 
+TEST_F(TestDeviceArrayRoundtrip, Struct) {
```

Comment:
Should there also be a roundtrip with string/binary data?

Reply:
The struct data being used in this round trip is using a string column, I confirmed that it hits the scenario for string types that i'm addressing here.

Reply:
Oops, sorry for letting this slip through. Yes, this is sufficient indeed!